src/um_fields_subset.py

```diff
@@ -125,153 +82,126 @@ def create_default_outname(filename, suffix="_subset"):
     output_filename: str
         The default output filename.
     """
+
     output_filename = f"{filename}{suffix}"
     num=""
     if os.path.exists(output_filename):
         num = 1
         while os.path.exists(f"{output_filename}{num}"):
             num += 1
     return f"{output_filename}{num}"
-
-
-def filter_fields(input_file, prognostic, include_list, exclude_list):
+    
+def field_not_present_warning(fields, stash_list):
     """
-    Creates a list of fields from the input file that should be included based on user conditions.
-
+    Checks that the fields in the either list provided are present in the file and provides a warning if not.
+    
     Parameters
-    ----------
-    field : object
-        The field object to be checked.
-    prognostic : bool
-        A boolean flag indicating if only prognostic fields should be included.
-    include_list : list of int
-        A list of STASH item codes to include.
-    exclude_list : list of int
-        A list of STASH item codes to exclude.
+    __________
 
-    Returns
-    -------
-    list
-        A list of fields that need to be in the file
-    """
-    filtered_fields = []
-    for field in input_file.fields:
+    fields : mule.ff.FieldsFile.fields
+        All the fields from the fields file.
 
-        # Check if the field is part of the exclusion list
-        if field.stash in exclude_list:
-            continue
-            
-        # Check if the field is part of the inclusion list (if specified)
-        if include_list  and field.stash in include_list:
-            filtered_fields.append(field)
+    stash_list :  list of int
+        Either the exclude or include STASH item code list.
+    Returns
+    _______
+        None - gives a warning.
 
-        # If no inclusion or exclusion list, include the field based on its type
-        elif prognostic and field.lbuser4 in PROG_STASH_CODES:
-            filtered_fields.append(field)
+    """
 
-        elif not prognostic and not include_list and not exclude_list:
-            filtered_fields.append(field)
+    existing_codes = {f.lbuser4 for f in fields}
+    missing_codes = [code for code in stash_list if code not in existing_codes]
```

Comment:
```suggestion
    missing_codes = stash_list - existing_codes
```

You can use subtraction between python sets here to find those present in `stash_list` and not in `existing_codes`.